cpp/src/ray/util/function_helper.cc

```diff
@@ -1,94 +1,173 @@
+  auto it = libraries_.find(lib_path.string());
   if (it != libraries_.end()) {
-    return it->second;
+    return;
   }
 
-  auto absolute_path = boost::filesystem::absolute(lib_name);
-  if (!boost::filesystem::exists(absolute_path)) {
-    RAY_LOG(WARNING) << lib_name << " dll not found, absolute_path: " << absolute_path
-                     << ", current path" << boost::filesystem::current_path();
-    return nullptr;
+  if (!boost::filesystem::exists(lib_path)) {
+    RAY_LOG(WARNING) << lib_path << " dynamic library not found.";
+    return;
   }
 
   std::shared_ptr<boost::dll::shared_library> lib = nullptr;
   try {
     lib = std::make_shared<boost::dll::shared_library>(
-        lib_name, boost::dll::load_mode::type::rtld_lazy);
+        lib_path.string(), boost::dll::load_mode::type::rtld_lazy);
   } catch (std::exception &e) {
-    RAY_LOG(WARNING) << "Load library failed, lib_name: " << lib_name
+    RAY_LOG(WARNING) << "Load library failed, lib_path: " << lib_path
                      << ", failed reason: " << e.what();
+    return;
   } catch (...) {
-    RAY_LOG(WARNING) << "Load library failed, lib_name: " << lib_name
+    RAY_LOG(WARNING) << "Load library failed, lib_path: " << lib_path
                      << ", unknown failed reason.";
+    return;
   }
 
-  if (lib == nullptr) {
-    return nullptr;
-  }
+  RAY_LOG(INFO) << "Loaded library: " << lib_path << " successfully.";
+  RAY_CHECK(libraries_.emplace(lib_path.string(), lib).second);
 
-  RAY_LOG(INFO) << "Loaded library: " << lib_name << " successfully.";
-  RAY_CHECK(libraries_.emplace(lib_name, lib).second);
-  return lib;
+  try {
+    auto entry_func = boost::dll::import_alias<msgpack::sbuffer(
+        const void *, const std::string &, const std::vector<msgpack::sbuffer> &,
+        msgpack::sbuffer *)>(*lib, "TaskExecutionHandler");
+    auto function_names = LoadAllRemoteFunctions(lib_path.string(), *lib);
+    if (function_names.empty()) {
+      RAY_LOG(WARNING) << "No remote functions in library " << lib_path
+                       << ", maybe it's not a dynamic library of Ray application.";
+      lib->unload();
+      return;
+    }
+    RAY_LOG(INFO) << "The lib path: " << lib_path
+                  << ", all remote functions: " << function_names;
+    entry_funcs_.emplace(lib_path.string(), entry_func);
+    return;
+  } catch (std::exception &e) {
+    RAY_LOG(WARNING) << "Get execute function failed, lib_path: " << lib_path
+                     << ", failed reason: " << e.what();
+  } catch (...) {
+    RAY_LOG(WARNING) << "Get execute function failed, lib_path: " << lib_path
+                     << ", unknown failed reason.";
+  }
+  return;
 }
 
-std::string FunctionHelper::OutputAllRemoteFunctionNames(
-    const boost::dll::shared_library &lib) {
-  auto get_names_func =
-      boost::dll::import_alias<std::vector<std::string>()>(lib, "GetRemoteFunctionNames");
+std::string FunctionHelper::LoadAllRemoteFunctions(
```

Comment:
LoadAllRemoteFunctions. -> GetAllRemoteFunctionNames

Reply:
I think `LoadAllRemoteFunctions` is more reasonable because this function will load the function maps from the library. And the return value of function names is secondary.